### ISSUE TITLE
BUG: Error with invalid message is thrown

### ISSUE THREAD

**Issue by Comoque1** [cherrytwist/Server#2342]
**Describe the bug**

**To Reproduce**
1. Go to https://dev.alkem.io/graphql
2. Authenticate with Challenge Member
3. Perform the following mutation:
```
mutation updatePreferenceOnChallenge(
  $preferenceData: UpdateChallengePreferenceInput!
) {
  updatePreferenceOnChallenge(preferenceData: $preferenceData) {
    id
  }
}

{
  "preferenceData": {
    "challengeID": "622e508a-457c-4df6-a656-653c4ca61b4d",
    "type": "ALLOW_CONTRIBUTORS_TO_CREATE_OPPORTUNITIES",
    "value": "true"
  }
}
```
4. Error for organization privileges is returned instead of challenge: `"message": "Authorization: unable to grant 'update' privilege: organization preference update: 1e84a758-6bff-11ed-85b6-0277f22c4881",`
![image.png](https://images.zenhubusercontent.com/5f5ff9fd50a1a94a1f25f32c/5e6d97b8-1f90-4d18-9524-9386b12529f2)

**Expected behavior**
If user doesn't have rights try to change challenge preferences, correct message is returned

**Comment by techsmyth** [comment]
@Comoque1 also fixed with #2341

**Comment by Comoque1** [comment]
Addressed with [this PR](https://app.zenhub.com/workspaces/alkemio-development-5ecb98b262ebd9f4aec4194c/issues/alkem-io/server/2343)
